cranelift/codegen/src/opts/icmp.isle

```diff
@@ -47,6 +47,27 @@
              (iconst _ (u64_from_imm64 1))))
       extend)
 
+;; Rewrite loose inequalities to strict inequalities:
+;; ule(x, c) == ult(x, c+1), for c != UMAX.
+(rule (simplify (ule (fits_in_64 (ty_int bty)) x (iconst cty (u64_from_imm64 c))))
+       (if-let $false (u64_eq c (ty_umax cty)))
```

Comment:
It might be nice to have "checked-add" and "checked-sub" constructors (we may already?) so that one could write

```
(if-let c_plus_1 (u64_checked_add_ty cty c 1))
```

what do you think?

Reply:
I think I prefer the current version